src/app/libp2p_helper/src/libp2p_helper/main.go

```diff
@@ -1209,6 +1211,43 @@ func (ap *findPeerMsg) run(app *app) (interface{}, error) {
 	return *maybePeer, nil
 }
 
+type setTelemetryDataMsg struct {
+	Data string `json:"data"`
+}
+
+func (m *setTelemetryDataMsg) run(app *app) (interface{}, error) {
+	app.P2p.TelemetryData = m.Data
+	return "setTelemetryData success", nil
+}
+
+type getPeerTelemetryDataMsg struct {
+	PeerID string `json:"peer_id"`
+}
+
+func (m *getPeerTelemetryDataMsg) run(app *app) (interface{}, error) {
+	// Open a "get telemetry" stream on m.PeerID,
+	// block until you can read the response, return that.
+	s, err := app.P2p.Host.NewStream(app.Ctx, peer.ID(m.PeerID), codanet.TelemetryProtocolID)
+	if err != nil {
+		app.P2p.Logger.Error("failed to open stream: ", err)
+		return nil, err
+	}
+
+	defer func() {
+		_ = s.Close()
+	}()
+
+	var data string
+	dec := json.NewDecoder(s)
```

Comment:
We should also change this to read a fixed at most a fixed number of bytes (8192 should be enough...)

Reply:
kk, i can change it to not be JSON and just read into a fixed size buffer then